src/NuGet.Core/NuGet.Packaging/PackageExtraction/ZipArchiveExtensions.cs

```diff
@@ -67,25 +68,78 @@ public static string SaveAsFile(this ZipArchiveEntry entry, string fileFullPath,
 
         public static void UpdateFileTimeFromEntry(this ZipArchiveEntry entry, string fileFullPath, ILogger logger)
         {
-            var attr = File.GetAttributes(fileFullPath);
+            Testable.Default.UpdateFileTimeFromEntry(entry, fileFullPath, logger);
+        }
 
-            if (!attr.HasFlag(FileAttributes.Directory) &&
-                entry.LastWriteTime.DateTime != DateTime.MinValue && // Ignore invalid times
-                entry.LastWriteTime.UtcDateTime <= DateTime.UtcNow) // Ignore future times
+        internal static void UpdateFileTime(string fileFullPath, DateTime dateTime)
+        {
+            Testable.Default.UpdateFileTimeEntry(fileFullPath, dateTime);
+        }
+
+        internal class Testable
+        {
+            public static Testable Default { get; } = new Testable(EnvironmentVariableWrapper.Instance);
+
+            internal Testable(IEnvironmentVariableReader environmentVariableReader)
             {
-                try
+                _updateFileTimeFromEntryMaxRetries = 9;
```

Comment:
Was this value arbitrary or intentional (if so, comment why)

Reply:
It's arbitrary. As mentioned in my discussion in Kartheek's comments, we have no data as for how long AV tends to keep files open, or what a reasonable retry period is, balancing needless retries when something other than AV has the file open and will never unlock it, vs needlessly failing when AV has the file open for a limited time and we would have succeeded if we waited longer.